### PR TITLE
Validate unique tx atx

### PR DESCRIPTION
**Pull request by gavraz**

### CODE REVIEW

File: sync/validation_queue.go [hunk 1]
```diff
@@ -100,6 +100,11 @@ func (vq *validationQueue) traverse(s *Syncer, blk *types.BlockHeader) error {
 
 func (vq *validationQueue) finishBlockCallback(s *Syncer, block *types.Block) func() error {
 	return func() error {
+		// validate unique tx atx
+		if err := validateUniqueTxAtx(block); err != nil {
```
Comment (almogdepaz) [inline sync/validation_queue.go]: do this before we validate the view so we don't validate the view of syntactically invalid blocks

Comment (almogdepaz) [reply sync/validation_queue.go]: you can do it in the traverse() method just before or after BlockSignedAndEligible check
(depends which one of them is faster)

File: sync/syncer.go [hunk 2]
```diff
@@ -384,12 +384,42 @@ func (s *Syncer) GetFullBlocks(blockIds []types.BlockID) []*types.Block {
 	return blocksArr
 }
 
+var errDupTx = errors.New("duplicate TransactionId in block")
+var errDupAtx = errors.New("duplicate AtxId in block")
+
+func validateUniqueTxAtx(b *types.Block) error {
```
Comment (almogdepaz) [inline sync/syncer.go:390]: init maps size to len of ids

File: sync/syncer.go [hunk 3]
```diff
@@ -384,12 +384,42 @@ func (s *Syncer) GetFullBlocks(blockIds []types.BlockID) []*types.Block {
...
+	return nil
+}
+
 func (s *Syncer) BlockSyntacticValidation(block *types.Block) ([]*types.AddressableSignedTransaction, []*types.ActivationTx, error) {
 	//block eligibility
 	if eligible, err := s.BlockSignedAndEligible(block); err != nil || !eligible {
 		return nil, nil, fmt.Errorf("block eligibiliy check failed - err %v", err)
 	}
 
+	// validate unique tx atx
```
Comment (almogdepaz) [inline sync/syncer.go:413]: i think that validating uniq txids/atxids is faster than BlockSignedAndEligible, we should put it first in the flow

File: sync/syncer.go (hunk 2 above)
Comment (gavraz) [reply sync/syncer.go:390]: Done

File: sync/syncer.go (hunk 3 above)
Comment (gavraz) [reply sync/syncer.go:413]: Done

File: sync/validation_queue.go (hunk 1 above)
Comment (gavraz) [reply sync/validation_queue.go]: Done. 
@almogdepaz do you think I should add any prints there?

**Review by almogdepaz — APPROVED** [review]

**Pull request merged**